### ISSUE TITLE
Inconsistency of the  rmn and AA_R data, is this a bug?

### ISSUE THREAD

**Issue by goodluck1982** [wannier-developers/wannier90#302]
I want to reproduce the Berry curvature and optical conductivity of postw90.x by hand through reading seedname_hr.dat and seedname_r.dat file. I took monolayer MoS2 as example. However, I find the data I calculated manually do not coincide with the data calculated by postw90.x. Concretely, BCz (z component of Berry curvature) and Sxx (xx component of optical conductivity) are consistent with the results of postw90, but BCx, BCy, and Szz are different from the results of postw90 obviously. See the figures below.  I find that the differences originate from the  different rmn=<um|r|un> data  used.  The data written in seedname_r.dat are different from the AA_R data used in postw90 (subroutine get_AA_R in get_oper.F90). Then I output the AA_R data used by postw90. After recalculating using the AA_R data I got all data consistent with the results of postw90.

Is this a bug?  And which results are more reasonable, the data calculated using seedname_r.dat or using the AA_R of postw90?

Berry curvature Omega:
![MoS2-curv](https://user-images.githubusercontent.com/48718961/70812769-37e96800-1e03-11ea-8d6b-3d93567aaa12.png)

Optical conductivity sigma:
![MoS2-sigma](https://user-images.githubusercontent.com/48718961/70812772-37e96800-1e03-11ea-9668-9badc9f23cb5.png)

**Comment by stepan-tsirkin** [comment]
Dear goodluck!

What value of parameter use_ws_distance are you using?

**Comment by goodluck1982** [comment]
To be simple for comparison, I used  use_ws_distance=F.
In fact, because I used k mesh up to 10x10x1 in DFT calculation, I don't think use_ws_distance=T will make noticeable difference.

**Comment by giovannipizzi** [comment]
Hi!
Thanks for the report.
In case you already resolved this issue offline, could you please report back?
Otherwise, I'm pinging also @ivosouza0 for further comments.

**Comment by goodluck1982** [comment]
I only found the issue.
And I also found that the optical conductivity calculated using seedname_r.dat agrees better with the dielectric function from VASP.
However I'm not able to resolve it.

> Hi!
> Thanks for the report.
> In case you already resolved this issue offline, could you please report back?
> Otherwise, I'm pinging also @ivosouza0 for further comments.

**Comment by matthew7879** [comment]
The wannier90_r.dat gives the position operator matrix elements of r_x, r_y and r_z, but it must satisfy the relation:

( [ r_{\alpha} (a,b,c) ]{m,n} ) * = [ r{\alpha}(-a,-b,-c) ]_{n,m} where * is conjugate.

I must use these position operator matrix elements to construct the non-abelian Berry connection matrix. These Berry connection matrices must be Hermitian. But as the position operators matrix elements do not satisfy the above relation, that is if r_{\alpha} (a,b,c) ]{m,n} * != r{\alpha} (-a,-b,-c)]_{n,m} the Berry connection matrix will not be a Hermitian matrix. Has anyone found a way to resolve this issue or a workaround?

**Comment by jaemolihm** [comment]
Hi @goodluck1982 , this is now fixed on `develop` by #702 (v4.0.3).
  
The two files were using different finite-difference approximations, and they were both not quite accurate, especially for low-dimensional systems with few k points along some axis. #702 adds the `transl_inv_full` argument, which switches both `seedname_r.dat` and `postw90.x` (`AA_R`) to use the translationally equivariant formula, which was found to be more accurate than those previously available. In the reference, we show that the translationally equivariant formula is manifestly invariant under a rigid translation of the system, more accurate, and converges faster with k-mesh density.
                                           
In short, you should use
```
    write_rmn            = true
    transl_inv_full      = true
    write_ndegen_applied = true
```

Note that `write_ndegen_applied = true` is required when `use_ws_distance = true`. The file is then written on the expanded R list with the degeneracy weights divided out, and `seedname_wsvec.dat` must not be applied on top.
   
 Reference: J.-M. Lihm, M. Ghim, S.-J. Hong, and C.-H. Park, Phys. Rev. B (2026), [doi:10.1103/wv76-36ny](https://doi.org/10.1103/wv76-36ny), [arXiv:2604.22614](https://arxiv.org/abs/2604.22614).
